controllers/grid/queries/form/QueryForm.inc.php

```diff
@@ -226,21 +226,20 @@ function fetch($request, $template = null, $display = false, $actionArgs = array
 			if ($query->getStageId() == WORKFLOW_STAGE_ID_EXTERNAL_REVIEW || $query->getStageId() == WORKFLOW_STAGE_ID_INTERNAL_REVIEW) {
 
 				// Get all review assignments for current submission
-				$reviewAssignmentDao = DAORegistry::getDAO('ReviewAssignmentDAO'); /* @var $reviewAssignmentDao ReviewAssignmentDAO */
+				$reviewAssignmentDao = DAORegistry::getDAO('ReviewAssignmentDAO'); /** @var ReviewAssignmentDAO $reviewAssignmentDao */
 				$reviewAssignments = $reviewAssignmentDao->getBySubmissionId($query->getAssocId());
 
 				// Get current users roles
 				$assignedRoles = [];
 				$usersAssignments = $stageAssignmentDao->getBySubmissionAndStageId($query->getAssocId(), $query->getStageId(), null, $user->getId());
 				while ($usersAssignment = $usersAssignments->next()) {
-					$userGroupDao = DAORegistry::getDAO('UserGroupDAO'); /* @var $userGroupDao UserGroupDAO */
+					$userGroupDao = DAORegistry::getDAO('UserGroupDAO'); /** @var UserGroupDAO $userGroupDao */
 					$userGroup = $userGroupDao->getById($usersAssignment->getUserGroupId());
 					$assignedRoles[] = $userGroup->getRoleId();
 				}
-
-				// if current user is editor, add all reviewers
-				if ($user->hasRole([ROLE_ID_SITE_ADMIN], CONTEXT_SITE) || $user->hasRole([ROLE_ID_MANAGER], $context->getId()) ||
-						array_intersect([ROLE_ID_MANAGER, ROLE_ID_SUB_EDITOR], $assignedRoles)) {
+				
+				// if current user is editor/journal manager/site admin and not have author role , add all reviewers
+				if (($user->hasRole([ROLE_ID_SITE_ADMIN], CONTEXT_SITE) || $user->hasRole([ROLE_ID_MANAGER, ROLE_ID_SUB_EDITOR], $context->getId())) && empty(array_intersect($assignedRoles, [ROLE_ID_AUTHOR])) ) {
```

Comment:
@touhidurabir, I see the logic in adding the check to exclude users who are assigned with `ROLE_ID_AUTHOR`, but it looks like this change also turns the check for `ROLE_ID_MANAGER` and `ROLE_ID_SUB_EDITOR` from a check on `$assignedRoles` (which is about assignments to the submission) to a call to `$user->hasRole` (which is based on users being assigned to roles in the journal). Was that intentional?

Reply:
@asmecher yes that is intentional . It felt to me that any one who has `manager`, `site admin` or `editor` can  see associated users/reviewers, not just those who are associated with stage/submission and the exception is only when has an `author` role associated with that submission .  Now that you have mention this, wondering if I am wrong to such thoughts .

Reply:
@ajnyga might have a better sense than me; if I remember correctly he originally added the code to protect reviewer/author anonymization in cases where a user had multiple roles. @ajnyga, did you have a policy in mind for checking role assignments on the current submission vs. on the context?

Reply:
The logic around this is difficult, but we try to let stage assignments override context roles whenever possible. If a user is assigned to a submission, treat them only according to their assigned roles. If they are not assigned to a submission _in any role_, then we fall back to their context roles.

So I think the appropriate logic here, in plain terms, would be:

```php
if (
    array_intersect($assignedRoles, [ROLE_ID_MANAGER, ROLE_ID_SUB_EDITOR]) 
    || 
    (
        empty($assignedRoles)
        &&
        (
            $user->hasRole([ROLE_ID_MANAGER, $context->getId()])
            || 
            $user->hasRole([ROLE_ID_SITE_ADMIN, CONTEXT_SITE])
        )
    )
) {
```

Subeditors do not have any permissions on submissions when they are not assigned, so we don't include them in a `$user->hasRole()` check.

@touhidurabir would you like me to pick this up or do you have time to make the changes for me to review?

Reply:
> Subeditors do not have any permissions on submissions when they are not assigned, so we don't include them in a $user->hasRole() check.

@NateWr I was not aware of this particular check . In that sense your proposed check is the right approach as I did some manual testing . Updated it